streams/src/main/java/org/apache/kafka/streams/kstream/JoinWindows.java

```diff
@@ -31,7 +31,9 @@
  *     WHERE
  *       stream1.key = stream2.key
  *       AND
- *       stream2.ts - before <= stream1.ts <= stream2.ts + after
+ *       stream2.ts >= stream1.ts - before
+ *       AND
+ *       stream2.ts <= stream1.ts + after
```

Comment:
I'll replace it with the former.  I rather stay away from the later, as it is not obvious from `BETWEEN` whether the operator matches an open, close, or semi-open interval.